modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/Slice2DIndex.java

```diff
@@ -159,14 +175,17 @@ public synchronized Slice2DIndex getSlice2DIndex( int imageIndex ) throws IOExce
             long endDataPosition = raf.readLong();
 
             raf.seek(dataPosition);
-            int tIndex = raf.readInt();
-            int zIndex = raf.readInt();
+            int dimensions = raf.readInt();
```

Comment:
Hmmm, I guess I could store the dimension size as a negative value. If it can't find a negative value it assumes it is an old file and that the dimension size = 2